src/js/stores/MesosStore.js

```diff
@@ -0,0 +1,221 @@
+var EventEmitter = require("events").EventEmitter;
+
+var AppDispatcher = require("../AppDispatcher");
+var InfoStore = require("../stores/InfoStore");
+var InfoActions = require("../actions/InfoActions");
+var InfoEvents = require("../events/InfoEvents");
+var MesosActions = require("../actions/MesosActions");
+var MesosEvents = require("../events/MesosEvents");
+
+const FILES_TTL = 60000;
+const STATE_TTL = 60000;
+const MASTER_ID = "master";
+
+var stateMap = {};
+var taskFileMap = {};
+var taskFileRequestQue = [];
+
+var MesosStore = Object.assign({
+
+  getState: function (nodeId) {
+    var data = stateMap[nodeId];
+    if (data == null || data.timestamp + STATE_TTL < Date.now()) {
+      invalidateState(nodeId);
+      return null;
+    }
+    return data.state;
+  },
+
+  getTaskFiles: function (taskId) {
+    var data = taskFileMap[taskId];
+    if (data == null || data.timestamp + FILES_TTL < Date.now()) {
+      invalidateTaskFiles(taskId);
+      return null;
+    }
+    return data.files;
+  }
+
+}, EventEmitter.prototype);
+
+function addState(nodeId, state) {
+  stateMap[nodeId] = {
+    state: state,
+    timestamp: Date.now()
+  };
+  MesosStore.emit(MesosEvents.CHANGE);
+}
+
+function invalidateState(nodeId) {
+  stateMap[nodeId] = null;
+}
+
+function addTaskFiles(taskId, files) {
+  taskFileMap[taskId] = {
+    files: files,
+    timestamp: Date.now()
+  };
+  MesosStore.emit(MesosEvents.CHANGE);
+}
+
+function invalidateTaskFiles(taskId) {
+  delete taskFileMap[taskId];
+}
+
+function getNodeUrl(nodeId) {
+  var master = MesosStore.getState(MASTER_ID);
+  if (master == null) {
+    throw new Error(
```

Comment:
No, let's discuss it here, it seems there is a demand for discussion.
And I think you didn't got my point.
So, which valuable information do you mean? The user doesn't see the exception message at all. He only sees the crashed application, if you forgot to handle the exception correctly.
At the other hand, you do handling the exception always! So the message is completely meaningless, it will never be printed out, you just discard it.